SharedPackages/VPN/Sources/VPN/PacketTunnelProvider.swift

```diff
@@ -1368,28 +1376,34 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         completionHandler?(nil)
     }
 
-    @available(iOS 17, *)
+    /// Disables on-demand if the OS supports it.
+    ///
+    /// iOS 17+ supports it, but macOS does not - to the best of our knowledge.  Still, we encourage calling this method
+    /// in macOS too in case this feature is ever implemented.
+    ///
     @MainActor
-    public func handleShutDown(dueTo error: Error) async throws {
-        enum ShutDownError: Error {
-            case noManagerFound
+    public func disableOnDemandIfSupportedByOS() async throws {
+        enum DisableOnDemandError: LocalizedError {
+            case notSupportedByOS
+
+            func localizedDescription(in locale: Locale = .current) -> String {
+                switch self {
+                case .notSupportedByOS:
+                    return "Disabling on-demand is not supported by the OS"
+                }
+            }
         }
 
         Logger.networkProtection.log("🔴 Disabling Connect On Demand and shutting down the tunnel")
         let managers = try await NETunnelProviderManager.loadAllFromPreferences()
 
         guard let manager = managers.first else {
-            Logger.networkProtection.log("Could not find a viable manager, bailing out of shutdown")
-            // Doesn't matter a lot what error we throw here, since we'll try cancelling the
-            // tunnel.
-            throw TunnelError.vpnAccessRevoked(ShutDownError.noManagerFound)
+            throw DisableOnDemandError.notSupportedByOS
         }
 
         manager.isOnDemandEnabled = false
         try await manager.saveToPreferences()
         try await manager.loadFromPreferences()
-
-        await cancelTunnel(with: error)
```

Comment:
This method is `disableOnDemandIfSupportedByOS`... so it just tries to do that and if it's not possible it won't.  This is done by disabling on-demand.  The caller can decide on next steps if `disableOnDemandIfSupportedByOS` fails - so no need to do that here... this was a bit out of scope really.